src/lightning_app/utilities/app_helpers.py

```diff
@@ -483,3 +484,28 @@ def _load_state_dict(root_flow: "LightningFlow", state: Dict[str, Any], strict:
         for component_name in dynamic_components:
             if component_name not in components_names:
                 raise Exception(f"The component {component_name} was re-created during state reloading.")
+
+
+# Taken from https://github.com/MacHu-GWU/inspect_mate-project/blob/master/inspect_mate/tester.py#L117
+# Credits to @MacHu-GWU
+def is_static_method(klass_or_instance, attr):
```

Comment:
I think this is unnecessarily complicated. A static method is one that is decorated with @staticmethod and in the dict of the class itself. 

```py
def is_static_method(klass_or_instance, attr):
    klass = type(klass_or_instance) if not isinstance(klass_or_instance, type) else klass_or_instance
    return isinstance(klass.__dict__.get(attr), staticmethod)
```

Reply:
Let's also add unit tests for this.

Reply:
Done !